### DISCUSSION TITLE
create index failed, collection is loaded, please release it first

### DISCUSSION THREAD

**Discussion by zhenzi0322** [milvus-io/milvus#21347] [Ideas & Feature requests]
milvus2.2的版本后是不是不能每次添加向量时都添加索引？

**Comment by ingale726** [comment]
先添加索引，再添加向量吧

**Reply by zhenzi0322** [reply]
我就是这样操作，不知道为什么添加索引就会出错，不创建索引就没有出错。

**Reply by ingale726** [reply]
索引是创建空集合后创建的，之后就是添加向量，不用每添加一个向量就创建一次索引

**Reply by zhenzi0322** [reply]
> 索引是创建空集合后创建的，之后就是添加向量，不用每添加一个向量就创建一次索引

那就是说索引是只创建一次了，创建一个空集合后，再创建索引，最后就是不断地添加向量，添加向量过程中不需要创建索引了是吧

**Reply by zhenzi0322** [reply]
就是说集合跟索引是只创建一次的，是不?

**Reply by xiaofan-luan** [reply]
是的，索引只需要创建一次

**Reply by ingale726** [reply]
> > 索引是创建空集合后创建的，之后就是添加向量，不用每添加一个向量就创建一次索引
> 
> 那就是说索引是只创建一次了，创建一个空集合后，再创建索引，最后就是不断地添加向量，添加向量过程中不需要创建索引了是吧

是的

**Reply by zhenzi0322** [reply]
好的，我现在知道了。谢谢了

**Comment by ingale726** [comment]
对



在 2022-12-22 17:08:34，"振子" ***@***.***> 写道：

就是说集合跟索引是只创建一次的，是不?
